csharp/src/Google.Protobuf/MessageExtensions.cs

```diff
@@ -140,6 +143,75 @@ public static ByteString ToByteString(this IMessage message)
             return ByteString.AttachBytes(message.ToByteArray());
         }
 
+        /// <summary>
+        /// Checks if all required fields in a message have values set. For proto3 messages, this returns true
+        /// </summary>
+        public static bool IsInitialized(this IMessage message)
+        {
+            if (message.Descriptor.File.Proto.Syntax != "proto2")
+            {
+                return true;
+            }
+
+            return message.Descriptor
+                .Fields
+                .InDeclarationOrder()
+                .Where(f => f.IsRequired || f.IsRepeated || f.IsMap || f.MessageType != null)
+                .All(f =>
```

Comment:
I am not an expert for C#, does this return true when all checks bellow return true, and it return false when one or more checks bellow return false?

Reply:
Yes